scripts/generate-locales.ts

```
@@ -284,21 +284,39 @@ async function updateLocaleFileHook(
  *
  * @param filePath The full file path to the file.
  * @param definitionKey The definition key of the current file (ex. 'location').
+ * @param locale The locale for that file.
  */
-async function normalizeLocaleFile(filePath: string, definitionKey: string) {
+async function normalizeLocaleFile(
+  filePath: string,
+  definitionKey: string,
+  locale: string
+) {
   function normalizeDataRecursive<T>(localeData: T): T {
     if (typeof localeData !== 'object' || localeData === null) {
       // we can only traverse object-like structs
       return localeData;
     }
 
+    let collator = null;
+    try {
+      // eslint-disable-next-line no-restricted-globals
+      collator = new Intl.Collator(locale.replaceAll('_', '-'));
+    } catch {
+      if (locale == 'base') {
+        // eslint-disable-next-line no-restricted-globals
+        collator = new Intl.Collator('en');
+      } else {
+        throw(`Failed to create collator for locale ${locale}. Using default collator.`);
```

Please throw a proper error, adjust the message and include the cause.

---

no longer needed.